src/Models.js

```diff
@@ -33,14 +34,50 @@
       this.expInThisLevel = levelExp;
       this.expTillNextLevel = totalExpNeeded;
     }
+  }
+
+  class Timer{
+    constructor(timerData){
+      this.history = timerData.history;
+      this.startTime = timerData.startTime;
+    }
+
+    startWork(workType) {
+      this.startTime = new Date().getTime();
+      this.history[this.startTime] = {type: workType, finishTime: 0};
+    }
+
+    finishWork() {
+      const finishTime = new Date().getTime();
+      this.history[this.startTime].finishTime = finishTime;
+      const type = this.history[this.startTime].type;
+      const timeWorked = finishTime - this.startTime;
+      this.startTime = 0;
+      return timeWorked;
+    }
+  }
 
+  LifeGamification.models.startWork = function (skill, type) {
+    return new Promise((resolve, reject) => {
+      skill.timer.startWork(type);
+      saveSkillsCollection()
+        .then(resolve);
+    });
+  }
+
+  LifeGamification.models.finishWork = function (skill) {
+    return new Promise((resolve, reject) => {
+      const addedExp = Math.floor(skill.timer.finishWork() / 60000);
+      LifeGamification.models.updateExp(skill, addedExp)
+        .then(resolve);
+    });
   }
 
   LifeGamification.models.createSkillsCollection = function (skillsJSON) {
     return new Promise((resolve, reject) => {
       for (let skillName in skillsJSON) {
         const skillData = skillsJSON[skillName];
-        const newSkill = new Skill(skillName, skillData.exp);
+        const newSkill = new Skill(skillName, skillData.exp, skillData.timer);
```

Comment:
What if the skillData doesn't contain `timer` key? Is JavaScript handling things like this?
We need to have code compatible with old versions of LG.

Reply:
Apparently yes. It will display "undefined".